09_00_mutable_references.md

```diff
@@ -172,23 +200,42 @@ bytes slice memory address after calling read_version: 0x50
 ```
 
-Notice how `transaction_bytes` starts with the same memory address as our slice in the `main` function and after being modified has a new memory address. Our `bytes_slice` in the `main` function should also have changed, but it does not and remains the same. We can see that the slice in the `read_version` function no longer returns the first 4 bytes, `[1, 0, 0, 0]` after being read, whereas the one in the `main` function still does.
-
-The slice in the `main` function is no longer the same as the one in the `read_version` function. What's happening here is that in the `read_version` function the local variable is being changed without modifying the slice in `main`. The `transaction_bytes` variable is being treated as a mutable *copy* and not a mutable *reference*. That's not what we want. We want to change the original slice in `main` as well. 
-
-What we really want to do is pass around a reference so that there is a link between the variable in `read_version` and the one in `main`. We can indicate that something is a mutable reference by prepending the `&mut` keyword. 
-
-Instead of the argument type being `mut transaction_bytes: &[u8]`, we want the the `transaction_bytes` variable to be of the type `&mut &[u8]`. 
+The specific addresses you get will probably be different from the snippet above.
+Notice how `transaction_bytes` starts with the same memory address as our slice in the `main` function and after being modified has a new memory address.
+Our `bytes_slice` in the `main` function should also have changed, but it does not and remains the same.
+We can see that the slice in the `read_version` function no longer returns the first 4 bytes, `[1, 0, 0, 0]` after being read, whereas the one in the `main` function still does.
+
+The slice in the `main` function is no longer the same as the one in the `read_version` function.
+What's happening here is that in the `read_version` function the local variable is being changed without modifying the slice in `main`.
+The `transaction_bytes` variable is being treated as a mutable *copy* of the pointer to the underlying data.
+
+Let me be more verbose since this can be pretty confusing for newcomers.
```

Comment:
Hm given this long paragraph, might make sense to add another diagram in here. Let me look into creating one that could help.

Reply:
I was uncertain if you would approve this, but since you said there was some people with difficulty here I added anyway.